pkg/manifest/manifest.go

```diff
@@ -60,16 +97,24 @@ func (m *Manifest) UnmarshalJSON(in []byte) error {
 	if !ok {
 		return fmt.Errorf("expected manifest to decode into *unstructured.Unstructured, got %T", ud)
 	}
-
+	if resourceId, err := unmarshalResourceId(m.Raw); err != nil {
```

Comment:
I feel like we can skip the YAML unmarshal with something like:

```go
m.GVK = ud.GroupVersionKind()
m.Obj = ud
m.resourceId = resourceId{
  Group: m.GVK.Group,
  Kind: m.GVK.Kind,
  Namespace: m.Obj.GetNamespace(),
  Name: m.Obj.GetName(),
}
return validateResourceId(m.resourceId)
```

Reply:
Yes